polkadot/node/core/pvf/src/execute/queue.rs

```diff
@@ -608,3 +618,151 @@ pub fn start(
 	.run();
 	(to_queue_tx, from_queue_rx, run)
 }
+
+struct Unscheduled {
+	unscheduled: HashMap<ExecutePriority, VecDeque<ExecuteJob>>,
+	priority_log: VecDeque<ExecutePriority>,
```

Comment:
I think this can be implemented without the queue. We will count number of jobs dispatched per priority ( **A** - low, **B**- normal, **C** critical). Then, for every **N** jobs (in our case N = 10) received we reset the counters for each priority to 0. 

In your code we want to allow at least 10% of low priority jobs to be executed so `P = 10% * N, P > 0`. Meaning dispatch 1 low priority job out of every 10.

Then  `select_next_priority()` should return low priority only if:
- the normal/critical queues are empty
- `B + C == N - P`, meaning the normal/critical priority messages have used up all the allocated space.